openapi-manager/src/environment.rs

```diff
@@ -0,0 +1,346 @@
+// Copyright 2025 Oxide Computer Company
+
+//! Describes the environment the command is running in, and particularly where
+//! different sets of specifications are loaded from
+
+use crate::apis::ManagedApis;
+use crate::git::GitRevision;
+use crate::output::headers::GENERATING;
+use crate::output::headers::HEADER_WIDTH;
+use crate::output::Styles;
+use crate::spec_files_blessed::BlessedFiles;
+use crate::spec_files_generated::GeneratedFiles;
+use crate::spec_files_local::walk_local_directory;
+use crate::spec_files_local::LocalFiles;
+use anyhow::Context;
+use camino::Utf8Component;
+use camino::Utf8Path;
+use camino::Utf8PathBuf;
+use owo_colors::OwoColorize;
+
+#[derive(Clone, Debug)]
+pub struct Environment {
+    /// The command to run the OpenAPI manager.
+    pub(crate) command: String,
+
+    /// Path to the root of this repository
+    pub(crate) repo_root: Utf8PathBuf,
+
+    /// The default OpenAPI directory.
+    pub(crate) default_openapi_dir: Utf8PathBuf,
+
+    /// The default Git upstream.
+    pub(crate) default_git_branch: String,
+}
```

Comment:
These are passed in as part of the environment.

Reply:
Yeah, the default OpenAPI directory and default Git branch were hardcoded in Omicron. We ask users to pass it in instead.